core/src/main/scala/org/apache/pekko/persistence/r2dbc/ConnectionFactoryProvider.scala

```diff
@@ -62,13 +73,29 @@ class ConnectionFactoryProvider(system: ActorSystem[_]) extends Extension {
         configLocation => {
           val config = system.settings.config.getConfig(configLocation)
           val settings = new ConnectionFactorySettings(config)
-          createConnectionPoolFactory(settings)
+          val customizer = createConnectionFactoryOptionsCustomizer(settings)
+          createConnectionPoolFactory(settings, customizer, config)
         })
       .asInstanceOf[ConnectionFactory]
   }
 
-  private def createConnectionFactory(settings: ConnectionFactorySettings): ConnectionFactory = {
+  private def createConnectionFactoryOptionsCustomizer(
+      settings: ConnectionFactorySettings): ConnectionFactoryOptionsCustomizer = {
+    settings.connectionFactoryOptionsCustomizer match {
+      case None => NoopCustomizer
+      case Some(fqcn) =>
+        val args = List(classOf[ActorSystem[_]] -> system)
+        system.dynamicAccess.createInstanceFor[ConnectionFactoryOptionsCustomizer](fqcn, args) match {
+          case Success(customizer) => customizer
+          case Failure(cause) =>
+            throw new RuntimeException("Failed to create ConnectionFactoryOptionsCustomizer", cause)
```

Comment:
Could you make this an IllegalStateException and include the invalid FCQN value in the message?

Reply:
Fixed. Used the latter